lib/graphql/batch/query_callback.rb

```diff
@@ -0,0 +1,16 @@
+module GraphQL::Batch
+  class QueryCallback < QueryContainer
+    def initialize(query, &block)
+      super(query)
+      @block = block
+    end
+
+    def call(result, error)
+      begin
+        complete(@block.call(result, error))
+      rescue => block_error
```

Comment:
I'm wondering if this should be broader than `StandardError` or not, just in case someone tries to `.rescue(Exception)` for some reason? I know it's usually considered bad form to rescue from anything higher than `StandardError`, so just curious if that's still true for promise-y stuff like this?

(may be irrelevant if we end up using a promise library)